graphql/src/main/java/io/strimzi/admin/graphql/GraphQLService.java

```diff
@@ -0,0 +1,141 @@
+import io.strimzi.admin.graphql.registration.RuntimeWiringRegistry;
+import io.strimzi.admin.http.server.registration.RouteRegistration;
+import io.strimzi.admin.http.server.registration.RouteRegistrationDescriptor;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.graphql.GraphQLHandler;
+import io.vertx.ext.web.handler.graphql.GraphiQLHandler;
+import io.vertx.ext.web.handler.graphql.GraphiQLHandlerOptions;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * GraphQLService configures the GraphQL library by making an executable schema from a schema
+ * ({@link graphql.schema.idl.TypeDefinitionRegistry}) and its implementation
+ * ({@link graphql.schema.idl.RuntimeWiring}).
+ * <p>
+ * The executable schema is exposed on the /graphql endpoint of the http-server.
+ * If the vertx web development environment is set, the graphiql tool will also be made available
+ * as the /graphiql endpoint on the http-server.
+ * <p>
+ * The schema and its implementation are built in a modular fashion to allow the full executable
+ * schema to be decomposed into logical/business areas. Each module defines the schema and
+ * implementation relevant to its logical/business area and exposes this through a
+ * {@link GraphQLRegistration} object which is loaded from the classpath using the Java service
+ * loader. All the individual GraphQLRegistration objects are processed in this service and the
+ * schemas and implementations merged. The merged objects are then used to create the executable
+ * schema of the {@link graphql.GraphQL} which controls the processing of the GraphQL requests
+ * which are passed through the /graphql endpoint.
+ */
+public class GraphQLService implements RouteRegistration {
+
+    private static final Logger LOGGER = LogManager.getLogger(GraphQLService.class);
+    private static final String BASE_SCHEMA_LOCATION = "graphql-schema/baseSchema.graphql";
+
+    @Override
+    public Future<RouteRegistrationDescriptor> getRegistrationDescriptor(Vertx vertx) {
+        final Promise<RouteRegistrationDescriptor> promise = Promise.promise();
+
+        final Router router = Router.router(vertx);
+        final RouteRegistrationDescriptor graphQLRegistrationDescriptor = RouteRegistrationDescriptor.create("/", router);
+
+        if (GraphiQLHandlerOptions.DEFAULT_ENABLED) {
+            LOGGER.warn("GraphiQL is enabled");
+            router.route("/graphiql/*").handler(GraphiQLHandler.create());
+        }
+
+        configureGraphQLHandler(vertx)
+            .onSuccess(graphQLHandler -> {
+                graphQLHandler.queryContext(routingContext -> routingContext);
```

Comment:
what exactly this does? Just returning what it's in input?

Reply:
In graphql-java, the values for nodes are resolved using datafetchers (also referred to as resolvers). These decide what value should be returned for a particular type or property in a type. These datafetchers have access to a context which they can use in determining what result should be returned for fetching the type or property and graphql-java lets you set that context to any value you want to. In the strimzi-admin case, we just assign the Vert.x routingcontext so the datafetchers can get the routingcontext and the will allow us to pass a security object down to the datafetchers so that calls to the adminclient from the datafetchers can provide the necessary login string for authentication/authorization. You can look at the Vert.x docs on it here: https://vertx.io/docs/vertx-web-graphql/java/#_providing_data_fetchers_with_some_context